goaviatrix/client_proxy_config.go

```diff
@@ -0,0 +1,76 @@
+package goaviatrix
+
+import (
+	"strings"
+)
+
+type ClientProxyConfig struct {
+	HttpProxy          string `json:"http_proxy,omitempty"`
+	HttpsProxy         string `json:"https_proxy,omitempty"`
+	ProxyCaCertificate string
+}
+
+type ClientProxyConfigResp struct {
+	Return  bool
+	Results ClientProxyConfig
+	Reason  string
+}
+
+func (c *Client) CreateClientProxyConfig(clientProxyConfig *ClientProxyConfig) error {
+	action := "apply_proxy_config"
+	if clientProxyConfig.ProxyCaCertificate != "" {
+		params := map[string]string{
+			"CID":         c.CID,
+			"action":      action,
+			"http_proxy":  clientProxyConfig.HttpProxy,
+			"https_proxy": clientProxyConfig.HttpsProxy,
+		}
+
+		var files []File
+		ca := File{
+			ParamName:      "server_ca_cert",
+			UseFileContent: true,
+			FileName:       "ca.pem", // fake name for ca
+			FileContent:    clientProxyConfig.ProxyCaCertificate,
+		}
+		files = append(files, ca)
+		return c.PostFileAPI(params, files, BasicCheck)
+	} else {
+		data := map[string]interface{}{
+			"CID":         c.CID,
+			"action":      action,
+			"http_proxy":  clientProxyConfig.HttpProxy,
+			"https_proxy": clientProxyConfig.HttpsProxy,
+		}
+		return c.PostAPI(action, data, BasicCheck)
+	}
+}
```

Comment:
We can clean this up a bit by only defining the data map once and just adding the file to the []File inline
```
	action := "apply_proxy_config"
	params := map[string]string{
		"CID":         c.CID,
		"action":      action,
		"http_proxy":  clientProxyConfig.HttpProxy,
		"https_proxy": clientProxyConfig.HttpsProxy,
	}
	if clientProxyConfig.ProxyCaCertificate != "" {
		files := []File{
			{
				ParamName:      "server_ca_cert",
				UseFileContent: true,
				FileName:       "ca.pem", // fake name for ca
				FileContent:    clientProxyConfig.ProxyCaCertificate,
			},
		}
		return c.PostFileAPI(params, files, BasicCheck)
	} else {
		return c.PostAPI(action, params, BasicCheck)
	}
```